src/catalog/backends/common/src/main/java/org/geoserver/cloud/event/remote/resourcepool/RemoteEventResourcePoolProcessor.java

```diff
@@ -35,14 +33,14 @@
 @Slf4j(topic = "org.geoserver.cloud.event.remote.resourcepool")
 public class RemoteEventResourcePoolProcessor {
 
-    private Catalog rawCatalog;
+    private final GeoServerImpl rawGeoServer;
 
     /**
-     * @param rawCatalog used to evict cached live data sources from its {@link
+     * @param rawGeoServer used to evict cached live data sources (via the catalog) from its {@link
      *     Catalog#getResourcePool() ResourcePool}
      */
-    public RemoteEventResourcePoolProcessor(Catalog rawCatalog) {
-        this.rawCatalog = rawCatalog;
+    public RemoteEventResourcePoolProcessor(GeoServerImpl rawGeoServer) {
```

Comment:
Please restore the `rawCatalog` argument. With `GeoServer` as argument, `GeoServer.getCatalog()` will return the final catalog (the raw catalog will be wrapped on a couple decorators)

Reply:
Done in 33c46d24e75dd4a50af3d908695a1e739d75ce4b